magento/magento2
# gitignore everything in pub/media

### Description (*)
The current .gitignore file is not consistent as to how it deals with files uploaded from the admin area. Images/files uploaded are treated differently, based on which folder they are uploaded to. This means that images uploaded to existing folders will not be added to git, but images added to new folders  will be added to git. 

By this pull request, all files uploaded to pub/media are ignored, except those .htaccess files that exists as a part of Magento. 

The important change in this pull request is line 36, and the remaining changes are just to ensure that the .htaccess are still added to git. 

The following lines are no longer required, and are removed. 
/pub/media/favicon/*
/pub/media/logo/*
/pub/media/themes/*

The exceptions for .htaccess are done in 3 lines. 
!/pub/media/customer/
/pub/media/customer/*
!/pub/media/customer/.htaccess

1. First make an exception for the folder
2. Then ignore everything in the folder
3. Then an exception for the .htaccces

This structure is required because of the change in line 36

### Related Pull Requests

### Fixed Issues (if relevant)

### Manual testing scenarios (*)
1. Upload an image for each option in design configuration (favicon, Logo, Watermarks, Transactional email) and then you will see that the logo for transactional emails are added to git (because the folder is created upon upload), whereas the rest of the images are not added to git (because the folders already exists)

2. Using the wysiwyg editor, create a new folder under Storage Root, and add a image to this folder. This image will be added to git, but images uploaded to existing folders will not. 

- After this pull request, try the previous scenarios, and none of the images should be added to git. 

- All the htaccess files are still added to git. 

### Questions or comments
This pull request is based on the assumption that images uploaded by the user, shouldn't be kept in git, which current gitignore also suggests. 

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#32820: gitignore everything in pub/media
